includes/sanitizers/class-amp-img-sanitizer.php

```diff
@@ -368,38 +368,71 @@ private function adjust_and_replace_node( $node ) {
 	 */
 	private function maybe_add_lightbox_attributes( $attributes, $node ) {
 		$parent_node = $node->parentNode;
-		if ( ! ( $parent_node instanceof DOMElement ) || 'figure' !== $parent_node->tagName ) {
+		if ( ! ( $parent_node instanceof DOMElement ) || ! ( $parent_node->parentNode instanceof DOMElement ) ) {
 			return $attributes;
 		}
 
-		// Account for blocks that include alignment.
-		// In that case, the structure changes from figure.wp-block-image > img
+		// This should be true for links to the actual media file, but not for links to the attachment page.
+		$is_node_wrapped_in_media_file_link = (
+			'a' === $parent_node->tagName
+			&&
+			( 'figure' === $parent_node->tagName || 'figure' === $parent_node->parentNode->tagName )
+			&&
+			attachment_url_to_postid( $parent_node->getAttribute( 'href' ) )
```

Comment:
This ensures the link is to the 'Media File,' not the 'Attachment Page':

<img width="958" alt="link-to-attachment-page" src="https://user-images.githubusercontent.com/4063887/67057996-7242cb80-f118-11e9-8211-efdda9e74f19.png">